src/cascadia/TerminalSettingsModel/IInheritable.h

```diff
@@ -133,20 +150,57 @@ private:                                                                    \
                                                                             \
         /*no value was found*/                                              \
         return std::nullopt;                                                \
+    }                                                                       \
+    projectedType _get##name##OverrideSourceImpl() const                    \

```

Comment:
I tried merging this logic with `_get##name##Impl` above. It's basically the same thing. Problem is, `_##name` is only accessible by the impl type, not the projected type.

I guess I could change `GETSET_SETTING` to take in the impl type instead of the projected type. Then just cast it on the way out of `<name>OverrideSource`? I'll try that out.

Reply:
oh wait. That doesn't work either because we need to define the return type. Idk if there's a way to do `auto` magic. I'm leaving it as-is for now unless somebody can provide some guidance/ideas.

Reply:
This is why I wanted you to implement parent walking in the base class. Let's eat the duplication for now.

Reply:
Filed #9211 as a followup